waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -202,6 +217,43 @@ public DatabaseMapping databaseMapping(@NotNull String databaseName) throws NoSu
         "Waggle Dance error no database mapping available tried to map database '" + databaseName + "'");
   }
 
+  @Override
+  public void checkTable(String databaseName, String tableName, DatabaseMapping mapping)
+      throws NoSuchObjectException {
+    String databasePrefix = mapping.getDatabasePrefix();
+    String transformedDbName = mapping.transformInboundDatabaseName(databaseName);
+    if (!isTableAllowed(databasePrefix, transformedDbName, tableName)) {
+      throw new NoSuchObjectException(String.format("%s.%s table not found", databaseName, tableName));
```

Comment:
```suggestion
      throw new NoSuchObjectException(String.format("%s.%s table not found in any mappings", databaseName, tableName));
```
^ just to be clear that it's not a table not found in the metastore.

Reply:
hmm yeah that's maybe a good point we want to indicate is because of WaggleDance probably. This reminds me @andreeapad what happens if you want to create a table that isn't mapped? Does it get created and then not shown or is creation not allowed?

Reply:
Ok, I'm not sure whether we want to have the same message that Hive would give when the table doesn't exist.

Reply:
I think the same exception is good but a different message so users at least have some hint that this is possible to change

Reply:
@patduin  it should get created then not shown

Reply:
Ok, so I'll change the message. The new comments appeared only after I sent mine oops

Reply:
> Ok, so I'll change the message. The new comments appeared only after I sent mine oops

That's OK, we're all comfortable with eventual consistency here :)

Reply:
on the created: ok I don't there's an ideal solution here, should probably only happen in theory, generally only federated metastores will be mapped and they are mostly read only